### ISSUE TITLE
Removing empty paragraph is merging and is bad UX

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5-engine#1566]
![oct-01-2018 10-57-22](https://user-images.githubusercontent.com/1502228/46279218-cceaa300-c568-11e8-94c7-b23dfcba0e9f.gif)

As you can see, there is an empty paragraph, and then there is a heading below it. Then, a user removes the empty paragraph. But under the hood, this is realized as a merge. So the heading is merged into an empty paragraph and the header becomes a paragraph.

AFAIR, I requested this "action" to be a merge because of collaboration :D. So when one user types in an empty paragraph and the other clicks <kbd>Backspace</kbd> in it, the text that was written by the first user is not deleted.

However, outside of a collaboration, it looks really bad. We should do something about it. It looks like a bug.

**Comment by Reinmar** [comment]
I think this is a DUP of https://github.com/ckeditor/ckeditor5-typing/issues/146.

**Comment by scofalik** [comment]
I don't think so. This is a different scenario. The linked issue discusses which element should be merged into which. This issue discusses whether merge or remove should be used.

**Comment by Reinmar** [comment]
But you wrote this:

> However, outside of a collaboration, it looks really bad. We should do something about it. It looks like a bug.

It looks like a bug because of https://github.com/ckeditor/ckeditor5-typing/issues/146. The used operation is a secondary thing here.

**Comment by scofalik** [comment]
These two issues are not connected.

We don't want to merge in this scenario. When you press <kbd>Backspace</kbd> or <kbd>Delete</kbd> in an empty paragraph you don't think about merging, you think about removing that blank space above your content. Merge direction is not important here cause we don't want to merge to happen anyway. Thus, the used operation is not a secondary thing.

**Comment by scofalik** [comment]
From the top of my head, a solution could be implemented in `MergeOperation#_execute`. It could check, on execution, if the element-to-merge-into is empty. If so, it could simply remove it. But even if it was done on execution, it probably would have impact on OT.

**Comment by Reinmar** [comment]
We talked F2F and we kinda agree that:

1. This is a special case of https://github.com/ckeditor/ckeditor5-typing/issues/146. But, this special case may actually require a different way of handling than https://github.com/ckeditor/ckeditor5-typing/issues/146 so fixing https://github.com/ckeditor/ckeditor5-typing/issues/146 without taking this case into consideration may not resolve the issue reported by @scofalik.
2. Operations used to handle this are a secondary aspect (because this whole handling goes through `deleteContent()` and the UX of this method should be fixed). However, using merge will give better results in OT. So we may consider tripled `rename(leftBlock) + setattrs(leftBlock,rightBlock.getAttrs()) + merge` to perform a "merge right" kind of action.
